examples/apache_libcloud/public-cloud-modular_instance.py

```diff
@@ -0,0 +1,221 @@
+"""Script used to import not monitored servers from cloud into Cyberwatch and delete terminated cloud servers"""
+
+# Based on the Apache Libcloud API https://libcloud.apache.org/index.html
+# Tested on Google Compute Engine and AWS EC2
+
+# Prerequisites :
+# - Install libcloud with command "pip3 install apache-libcloud"
+# - Choose a driver (EC2/GCE) and set the corresponding variables (https://libcloud.readthedocs.io/en/stable/compute/drivers/)
+# - Set up your Cyberwatch API key in api.conf in the same folder as the script, for an example https://github.com/Cyberwatch/cyberwatch_api_toolbox#configuration
+# - SSH key file of servers to import named "id_rsa"
+# - If you are not using the default credentials for agentless connections configured in Cyberwatch, set up SERVER_LOGIN and/or WINRM_password
+# Notes :
+# - All servers will be imported with group "cloud_crawling"
+
+import argparse
+import os
+import sys
+import socket
+
+from configparser import ConfigParser
+from libcloud.compute.types import Provider
+from libcloud.compute.providers import get_driver
+from cbw_api_toolbox.cbw_api import CBWApi
+
+SSH_KEY_SERVERS = open(os.path.expanduser('id_rsa')).read()
+SERVER_LOGIN = ""
+WINRM_PASSWORD_SERVERS = ""
+
+# GCE
+GCE_EMAIL = ""
+GCE_FILE = ""
+GCE_PROJECT_ID = ""
+# EC2
+EC2_ACCESS_KEY = ""
+EC2_SECRET_KEY = ""
+EC2_REGION = ""
+
+def connect_api():
+    '''Connect to the API and test connection'''
+    conf = ConfigParser()
+    conf.read(os.path.join(os.path.abspath(
+        os.path.dirname(__file__)), '.', 'api.conf'))
+    global API  # pylint: disable=global-variable-undefined
+    API = CBWApi(conf.get('cyberwatch', 'url'), conf.get(
+        'cyberwatch', 'api_key'), conf.get('cyberwatch', 'secret_key'))
+
+    API.ping()
+
+
+def get_node():
+    '''Get list of available nodes and prompt user to choose'''
+    print("Which Cyberwatch node do you want to use to import?")
+    for node in API.nodes():
+        print("ID: {}, name: {}".format(node.id, node.name))
+    node_id = input("Enter ID of node to use : ")
```

Comment:
Prompt only if the user has multiple nodes

Reply:
and do no trust the user, store API.nodes in a variable and use the id only if it is in the list.